### PR TITLE
AO Integrals

### PR DESCRIPTION
**Pull request by jwaldrop107**
**Is this pull request associated with an issue(s)?**
No

**Description**
Brings back (some of) the integral related property types, along with related connections.

Technically r2g as is, but I don't want to merge yet until I figure out if there are more changes need for Integrals.

### CODE REVIEW

**Comment by jwaldrop107** [comment]
This is part proposal, part placeholder. If the changes in Chemist make this obsolete or necessitate more changes, that's fine.

**Comment by ryanmrichard** [comment]
I was aiming to have integrals look more like:
```
class Visitor {
    auto run(Kinetic braket){/* make kinetic engine */}   
};

class Libint : public ModuleBase {
    Libint() { satisfies_property_type<BraKet<AOs, OperatorBase, AOs>(); }

    auto run() {
         auto braket = unwrap_inputs(inputs);
         Visitor v;
         braket.the_operator().visit(v);
         v.engine().run(x,y,z);
    }
}
```

File: include/simde/evaluate_braket/evaluate_braket.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    return pluginplay::declare_input().add_field<const_braket_t>("BraKet");
+}
+
+template<typename BraKetType>
+TEMPLATED_PROPERTY_TYPE_RESULTS(EvaluateBraKet, BraKetType) {
+    return pluginplay::declare_result().add_field<type::tensor>(
+      "tensor representation");
+}
+
+using EKinetic =
```
Comment (ryanmrichard) [inline include/simde/evaluate_braket/evaluate_braket.hpp:38]: `EKinetic` sounds to me like the kinetic energy (which would be `EvaluateBraKet<BraKet<Determinant, T, Determinant>>`).

Comment (jwaldrop107) [reply include/simde/evaluate_braket/evaluate_braket.hpp]: Updated the typedef names.

**Review by ryanmrichard — APPROVED** [review]

**Comment by jwaldrop107** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/NWChemEx/SimDE/actions/runs/11332962327) **New version**:[v0.0.37](https://github.com/NWChemEx/SimDE/releases/tag/v0.0.37) **Changes**:[v0.0.36...v0.0.37](https://github.com/NWChemEx/SimDE/compare/v0.0.36...v0.0.37)

**Pull request merged**